src/sentry/event_manager.py

```diff
@@ -306,6 +306,12 @@ def save(self, project_id, raw=False, assume_normalized=False, start_time=None,
             self._data["project"] = int(project_id)
             job = {"data": self._data, "start_time": start_time}
             jobs = save_transaction_events([job], projects)
+
+            if not project.flags.has_transactions:
+                first_transaction_received.send_robust(
+                    project=project, event=job["event"], sender=Project
+                )
```

Comment:
Doesen't this mean that, for a project that ingests both errors and transactions, we will execute both `record_first_transaction` and `record_first_error` while now we are executing only one of the two (for the first event) ?

Reply:
No, `record_first_error` will never be called on a transaction. This is because we exit from the save function early for a transaction (on this line https://github.com/getsentry/sentry/blob/master/src/sentry/event_manager.py#L309) and so it will never trigger the `first_event_received` signal.

Reply:
But if you receive a transaction first, now you set the has_transaction flag.
What if after the first transaction I receive an error. I will get to this code:

https://github.com/getsentry/sentry/blob/master/src/sentry/event_manager.py#L478-L479

Does the record_first_transaction code ever updates the `first_event` attribute ?

Reply:
The code you linked is the only place project.first_event is updated. A transaction will never get there.

Reply:
Wait, I got this, So "today" transactions would not get to https://github.com/getsentry/sentry/blob/master/src/sentry/event_manager.py#L478-L479.

Which means this code already makes no sense today? https://github.com/getsentry/sentry/pull/25930/files#diff-02080fe8c79d65425cbae51a6bf6fea582c253e875386c6e9b3a7065b1dd10e1L114